keras/src/layers/reshaping/reshape.py

```diff
@@ -37,7 +39,21 @@ class Reshape(Layer):
 
     def __init__(self, target_shape, **kwargs):
         super().__init__(**kwargs)
-        self.target_shape = tuple(target_shape)
+        target_shape = tuple(target_shape)
+        # test validity of target_shape
+        if target_shape.count(-1) > 1:
+            raise ValueError(
+                "The `target_shape` argument must not contain more than one "
+                "`-1` value. Received: target_shape={}".format(target_shape)
+            )
+        self.target_shape = target_shape
+        # precalculate all values that might be required
+        self.need_explicit_shape_for_batch_size_None = (
+            target_shape.count(-1) == 1
+        )
+        self.new_size_no_minus_one = math.prod(
+            d for d in target_shape if d != -1
+        )
```

Comment:
Because you removed the `build` method, add `self.built = True` at the end of `__init__`.